Tavisca.Bootcamp.LanguageBasics.Exercise3/Program.cs

```diff
@@ -28,20 +31,161 @@ static void Main(string[] args)
             Console.ReadKey(true);
         }
 
-        private static void Test(int[] protein, int[] carbs, int[] fat, string[] dietPlans, int[] expected)
-        {
+        private static void Test(int[] protein, int[] carbs, int[] fat, string[] dietPlans, int[] expected)
+        {
             var result = SelectMeals(protein, carbs, fat, dietPlans).SequenceEqual(expected) ? "PASS" : "FAIL";
             Console.WriteLine($"Proteins = [{string.Join(", ", protein)}]");
             Console.WriteLine($"Carbs = [{string.Join(", ", carbs)}]");
-            Console.WriteLine($"Fats = [{string.Join(", ", fat)}]");
+            Console.WriteLine($"Fats = [{string.Join(", ", fat)}]");
             Console.WriteLine($"Diet plan = [{string.Join(", ", dietPlans)}]");
-            Console.WriteLine(result);
+            Console.WriteLine(result);
+        }
+
+        public static int[] SelectMeals(int[] protein, int[] carbs, int[] fat, string[] dietPlans)
+        {
+            int[] calories = new int[protein.Length], indexTrack;
+            string diet;
+            bool flag;
+            results = new int[dietPlans.Length];
+
+            for (int i=0; i<protein.Length; i++)
+            {
+                calories[i] = (5 * protein[i]) + (5 * carbs[i]) + (9 * fat[i]);
+            }
+
+            for(int i=0; i<dietPlans.Length; i++)
+            {
+                diet = dietPlans[i];
+                flag = false;
+                indexTrack = new int[]{ };
+
+                if (diet.Length == 0)
+                {
+                    results[i] = 0;
+                }
+                else
+                {
+                    for (int j = 0; j < diet.Length; j++)
+                    {
+                        switch (diet[j])
+                        {
+                            case 'P':
+                                indexTrack = ComputeIndexes(protein, indexTrack, "max");

```

Comment:
you can read about concepts of `const` and static readonly variables.
And see if you can use it here.
`max` and `min` are constants and could be declared as such instead of having hardcoded values

Reply:
you could also go with the the concepts of enum